backend/src/main/java/ai/verta/modeldb/batchProcess/OwnerRoleBindingUtils.java

```diff
@@ -0,0 +1,306 @@
+    migrateDatasetVersions();
+    LOGGER.info("DatasetVersions done migration");
+
+    LOGGER.info("Migration End");
+  }
+
+  private static void migrateExperiments() {
+    LOGGER.debug("Experiments migration started");
+    Long count = getEntityCount(ExperimentEntity.class);
+
+    int lowerBound = 0;
+    final int pagesize = 200;
+    LOGGER.debug("Total experiments {}", count);
+
+    Role ownerRole = roleService.getRoleByName(ModelDBConstants.ROLE_EXPERIMENT_OWNER, null);
+    while (lowerBound < count) {
+
+      try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+        Transaction transaction = session.beginTransaction();
+        CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+
+        CriteriaQuery<ExperimentEntity> criteriaQuery =
+            criteriaBuilder.createQuery(ExperimentEntity.class);
+        Root<ExperimentEntity> root = criteriaQuery.from(ExperimentEntity.class);
+
+        CriteriaQuery<ExperimentEntity> selectQuery =
+            criteriaQuery.select(root).orderBy(criteriaBuilder.asc(root.get("id")));
+
+        TypedQuery<ExperimentEntity> typedQuery = session.createQuery(selectQuery);
+
+        typedQuery.setFirstResult(lowerBound);
+        typedQuery.setMaxResults(pagesize);
+        List<ExperimentEntity> experimentEntities = typedQuery.getResultList();
+
+        if (experimentEntities.size() > 0) {
+          Set<String> userIds = new HashSet<>();
+          for (ExperimentEntity experiment : experimentEntities) {
+            userIds.add(experiment.getOwner());
+          }
+          LOGGER.debug("Experiments userId list : " + userIds);
+          if (userIds.size() == 0) {
+            LOGGER.warn("userIds not found for Experiments on page lower boundary {}", lowerBound);
+            lowerBound += pagesize;
+            continue;
+          }
+
+          // Fetch the experiment owners userInfo
+          Map<String, UserInfo> userInfoMap =
+              authService.getUserInfoFromAuthServer(userIds, null, null);
+          for (ExperimentEntity experimentEntity : experimentEntities) {
+            UserInfo userInfoValue = userInfoMap.get(experimentEntity.getOwner());
+            if (userInfoValue != null) {
+              try {
+                roleService.createRoleBinding(
+                    ownerRole,
+                    new CollaboratorUser(authService, userInfoValue),
+                    experimentEntity.getId(),
+                    ModelResourceEnum.ModelDBServiceResourceTypes.EXPERIMENT);
+              } catch (Exception e) {
+                e.printStackTrace();
+                LOGGER.error(e.getMessage());
+              }
+            } else {
+              LOGGER.error(
+                  "Experiment owner not found from UAC response list : experimentId - {} & userId - {}",
+                  experimentEntity.getId(),
+                  experimentEntity.getOwner());
+              new ModelDBException(
```

Comment:
you forgot to throw it

Reply:
is it used intentionally to just print error message?

Reply:
Yes, it is used intentionally to just print an error message.